core/src/main/java/hudson/model/FingerprintCleanupThread.java

```diff
@@ -71,6 +75,12 @@ private static FingerprintCleanupThread getInstance() {
     }
 
     public void execute(TaskListener listener) {
+        Object fingerprintStorage = FingerprintStorage.get();
+        if (!(fingerprintStorage instanceof FileFingerprintStorage)) {
+            LOGGER.fine("External fingerprint storage is configured. Skipping execution");
```

Comment:
is there no API for this in external or just not needed?

Reply:
@timja thanks for reviewing :) We currently do not have an API that can support fingerprint cleanup in external storages, something we will be tackling in the next few weeks :) 
As of now, we limit the execution of fingerprint cleanup only when the default file based storage is configured